server/keyspace/keyspace_group.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keyspace
+
+import (
+	"context"
+
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/storage/endpoint"
+	"github.com/tikv/pd/pkg/storage/kv"
+	"go.uber.org/zap"
+)
+
+const defaultKeyspaceGroupID = uint32(0)
+
+// GroupManager is the manager of keyspace group related data.
+type GroupManager struct {
+	ctx context.Context
+	// store is the storage for keyspace group related information.
+	store endpoint.KeyspaceGroupStorage
+}
+
+// NewKeyspaceGroupManager creates a Manager of keyspace group related data.
+func NewKeyspaceGroupManager(ctx context.Context, store endpoint.KeyspaceGroupStorage) *GroupManager {
+	return &GroupManager{
+		ctx:   ctx,
+		store: store,
+	}
+}
+
+// Bootstrap saves default keyspace group info.
+func (m *GroupManager) Bootstrap() error {
+	defaultKeyspaceGroup := &endpoint.KeyspaceGroup{
+		ID: defaultKeyspaceGroupID,
+		// TODO: define a user kind type
+		UserKind: "default",
+	}
+	err := m.saveKeyspaceGroup(defaultKeyspaceGroup)
+	// It's possible that default keyspace group already exists in the storage (e.g. PD restart/recover),
+	// so we ignore the ErrKeyspaceGroupExists.
+	if err != nil && err != ErrKeyspaceGroupExists {
+		return err
+	}
+
+	return nil
+}
+
+// CreateKeyspaces creates keyspace groups.
+func (m *GroupManager) CreateKeyspaces(keyspaceGroups []*endpoint.KeyspaceGroup) error {
+	for _, keyspaceGroup := range keyspaceGroups {
+		// TODO: add replica count
```

Comment:
How will it pass into?

Reply:
The replica count is the input contained in the request, we need to add the member info to it according to the count.